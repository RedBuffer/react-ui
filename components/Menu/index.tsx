import Menu from './Menu';
import MenuItem from './MenuItem';

type Type = typeof Menu;

export interface ComponentInterface extends Type {
  MenuItem: typeof MenuItem;
}

const Component: ComponentInterface = Menu as ComponentInterface;

Component.MenuItem = MenuItem;

export { default as Menu } from './Menu';
export { default as MenuItem } from './MenuItem';

export default Component;
