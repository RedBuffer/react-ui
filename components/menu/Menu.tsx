import * as React from 'react';
import classnames from 'classnames';

export interface PropTypes {
  className?: string;
  selected?: boolean;
  children?: React.ReactNode;
  style?: React.CSSProperties;
}

const Component = ({
  style,
  children,
  className,
  ...restProps
}: PropTypes): React.ReactElement => {
  return (
    <div
      className={classnames(
        'px-0',
        'py-2',
        'shadow',
        'bg-white',
        'box-border',
        'overflow-x-auto',
        'overflow-y-hidden',
        'whitespace-no-wrap',
        className,
      )}
      style={style}
      {...restProps}
    >
      <ul className={classnames('list-none', 'p-0', 'm-0')}>{children}</ul>
    </div>
  );
};

export default Component;
