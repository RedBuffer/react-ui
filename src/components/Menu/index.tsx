import Menu from './Menu';
import MenuItem from './MenuItem';

type Type = typeof Menu;

export interface ComponentInterface extends Type {
  MenuItem: typeof MenuItem;
}

const Component: ComponentInterface = Menu as ComponentInterface;

Component.MenuItem = MenuItem;

export default Component;
