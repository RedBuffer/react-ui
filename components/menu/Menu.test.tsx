import * as React from 'react';
import { shallow, mount } from 'enzyme';

import Menu from './index';

describe('Menu', () => {
  let handleClick;
  let wrapper;
  const callHandleEdit = (item) => (e) => {
    e.stopPropagation();
    handleClick(item);
  };

  beforeEach(() => {
    handleClick = jest.fn((item) => item);
    wrapper = mount(
      <Menu>
        <Menu.Item>
          <a id="item-1" onClick={callHandleEdit('item 1')}>
            Item 1
          </a>
        </Menu.Item>
        <Menu.Item>
          <a id="item-2" onClick={callHandleEdit('item 2')}>
            Item 2
          </a>
        </Menu.Item>
        <Menu.Item>
          <a id="item-3" onClick={callHandleEdit('item 3')}>
            Item 3
          </a>
        </Menu.Item>
      </Menu>,
    );
  });

  it('should render correctly', () => {
    const component = shallow(<Menu />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with menu items', () => {
    const component = wrapper;

    expect(component).toMatchSnapshot();
  });

  it('click on menu item 1 ', () => {
    wrapper.find('#item-1').hostNodes().simulate('click');

    expect(handleClick.mock.calls[0][0]).toBe('item 1');
  });

  it('click on menu item 2 ', () => {
    wrapper.find('#item-2').hostNodes().simulate('click');

    console.log(handleClick.mock.calls);
    expect(handleClick.mock.calls[0][0]).toBe('item 2');
  });
});
