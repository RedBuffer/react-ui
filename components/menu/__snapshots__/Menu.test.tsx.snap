// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<div
  className="px-0 py-2 shadow bg-white box-border overflow-x-auto overflow-y-hidden whitespace-no-wrap"
>
  <ul
    className="list-none p-0 m-0"
  />
</div>
`;

exports[`Menu should render correctly with menu items 1`] = `
<Component>
  <div
    className="px-0 py-2 shadow bg-white box-border overflow-x-auto overflow-y-hidden whitespace-no-wrap"
  >
    <ul
      className="list-none p-0 m-0"
    >
      <Component>
        <li
          className="m-0 px-4 py-2 relative list-none btn-light react-ui-menu-item"
        >
          <a
            id="item-1"
            onClick={[Function]}
          >
            Item 1
          </a>
        </li>
      </Component>
      <Component>
        <li
          className="m-0 px-4 py-2 relative list-none btn-light react-ui-menu-item"
        >
          <a
            id="item-2"
            onClick={[Function]}
          >
            Item 2
          </a>
        </li>
      </Component>
      <Component>
        <li
          className="m-0 px-4 py-2 relative list-none btn-light react-ui-menu-item"
        >
          <a
            id="item-3"
            onClick={[Function]}
          >
            Item 3
          </a>
        </li>
      </Component>
    </ul>
  </div>
</Component>
`;
